esp/src/test-ui/tests/framework/SetupForDev.md

```diff
@@ -0,0 +1,127 @@
+This documentation provides a comprehensive guide to setting up an Ubuntu VM on Oracle VirtualBox, installing the HPCC-Platform, and preparing the environment for testing and development.
+
+#### Setting Up an Ubuntu VM on Oracle VirtualBox
+
+**Overview of the installation process on a Windows machine:**
+
+1. **Download and install Oracle VirtualBox:**
+    - Visit the Oracle VirtualBox [website](https://www.virtualbox.org/) and download the latest version for Windows.
+    - Follow the installation instructions to install VirtualBox on your system.
+
+2. **Download the Ubuntu 22.04 Desktop ISO file:**
+    - Go to the Ubuntu [downloads page](https://ubuntu.com/download/desktop) and download the Ubuntu 22.04 LTS ISO file.
+
+3. **Set up a new VM in VirtualBox:**
+    - Open VirtualBox and click on `New` to create a new virtual machine.
+    - Name the VM and select the type and version (Linux, Ubuntu 64-bit).
+    - Configure system settings such as memory size and hard disk (create a virtual hard disk now).
+    - Assign multiple CPUs to your VM if your host machine has a multi-core processor. A good starting point is to allocate 2-4 CPUs to the VM.
+    - Allocate memory based on the number of CPUs assigned. A good rule of thumb is to assign at least 1 GB of RAM per CPU. If you allocate 2 CPUs, assign at least 2 GB of RAM. For 4 CPUs, assign at least 4 GB of RAM, and so on. Depending on the tasks you'll be performing on the VM, you might need to allocate more memory.
+    - Link the ISO file by going to the `Settings` of the VM, navigating to `Storage`, and attaching the ISO file to the optical drive.
+    - Boot the VM and follow the installation wizard to complete the Ubuntu setup.
+
+#### Installing HPCC-Platform on the VM
+
+**After successfully installing the VM, proceed with installing the HPCC-Platform:**
+
+1. **Download the HPCC-Platform package:**
+   ```sh
+   wget https://cdn.hpccsystems.com/releases/CE-Candidate-9.8.2/bin/platform/hpccsystems-platform-community_9.8.2-1jammy_amd64_withsymbols.deb
+   ```
+
+2. **Install the package:**
+   ```sh
```

Comment:
This should show how to install the package downloaded and use a version of 9.8.x as an example

Reply:
Done.